src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/FileStorageService.cs

```diff
@@ -40,34 +46,53 @@ public FileStorageService(IConfiguration configuration, ILogger<FileStorageServi
         public async Task CopyReleaseFilesToPublicContainer(CopyReleaseFilesCommand copyReleaseFilesCommand)
         {
             var privateContainer =
-                await FileStorageUtils.GetCloudBlobContainerAsync(_privateStorageConnectionString,
+                await GetCloudBlobContainerAsync(_privateStorageConnectionString,
                     PrivateFilesContainerName);
             var publicContainer =
-                await FileStorageUtils.GetCloudBlobContainerAsync(_publicStorageConnectionString,
+                await GetCloudBlobContainerAsync(_publicStorageConnectionString,
                     PublicFilesContainerName);
 
-            var sourceDirectoryPath = AdminReleaseDirectoryPath(copyReleaseFilesCommand.ReleaseId);
-            var destinationDirectoryPath =
-                PublicReleaseDirectoryPath(copyReleaseFilesCommand.PublicationSlug, copyReleaseFilesCommand.ReleaseSlug);
-
-            if (copyReleaseFilesCommand.ReleaseSlug != copyReleaseFilesCommand.PreviousReleaseSlug)
+            // Slug may have changed for the amendment so also remove the previous contents if it has
+            if (copyReleaseFilesCommand.ReleaseSlug != copyReleaseFilesCommand.PreviousVersionSlug)
             {
                 var previousDestinationDirectoryPath =
                     PublicReleaseDirectoryPath(copyReleaseFilesCommand.PublicationSlug,
-                        copyReleaseFilesCommand.PreviousReleaseSlug);
+                        copyReleaseFilesCommand.PreviousVersionSlug);
                 await DeleteBlobsAsync(publicContainer, previousDestinationDirectoryPath);
             }
+            
+            var destinationDirectoryPath =
+                PublicReleaseDirectoryPath(copyReleaseFilesCommand.PublicationSlug, copyReleaseFilesCommand.ReleaseSlug);
 
             await DeleteBlobsAsync(publicContainer, destinationDirectoryPath);
 
-            await CopyDirectoryAsyncAndZipFiles(sourceDirectoryPath, destinationDirectoryPath, privateContainer,
-                publicContainer, copyReleaseFilesCommand,
-                (source, destination) => CopyFileUnlessBatchedOrMeta(source, copyReleaseFilesCommand.ReleaseId));
+            var versions = copyReleaseFilesCommand.ReleaseFileReferences.GroupBy(rfr => rfr.ReleaseId)
```

Comment:
Is this just getting unique ReleaseIds from the list of ReleaseFileReferences?  If so it might read easier to do:

`
var versions = copyReleaseFilesCommand.ReleaseFileReferences.Select(r => r.ReleaseId).Distinct();
`

Reply:
Done - cheers.